scripts/file_collector.py

```diff
@@ -0,0 +1,187 @@
+)
+from p2a_impacts.utils import get_region, REGIONS
+
+
+def setup_logging(log_level):
+    formatter = logging.Formatter(
+        "%(asctime)s %(levelname)s: %(message)s", "%Y-%m-%d %H:%M:%S"
+    )
+    handler = logging.StreamHandler()
+    handler.setFormatter(formatter)
+    logger = logging.getLogger("scripts")
+    logger.addHandler(handler)
+    logger.setLevel(getattr(logging, log_level))
+    return logger
+
+
+logger = setup_logging("INFO")
+
+
+def file_collection(csv, date_range, region, ensemble, connection_string):
+    """
+    Builds the variables that would be used in in p2a_impacts.resolve_rules
+    to the point of accessing the climate explorer database, but instead of evaluating
+    a rule it writes the paths for the files used to a text file.
+    """
+
+    # read csv
+    logger.info("Reading {}".format(csv))
+    rules = read_csv(csv)
+
+    # create parse tree dictionary and gather unique variables
+    logger.info("Building parse tree")
+    parse_trees = {}
+    variables = {}
+    region_variable = None
+    for rule, condition in rules.items():
+        try:
+            parse_trees[rule], vars, region_var = build_parse_tree(condition)
+        except SyntaxError as e:
+            logger.warning("{}, rule will be excluded".format(e))
+            continue
+
+        # check region var
+        if region_var:
+            region_variable = region_var
+
+        # add unique variables to set
+        for name, values in vars.items():
+            if name not in variables.keys():
+                variables[name] = values
+
+    # get values for all variables we would need if we were doing an evaluation
+    logger.info("Collecting variables")
+    Session = sessionmaker(create_engine(connection_string))
+    sesh = Session()
+
+    # write file paths for each variable to text file
+    for name, values in variables.items():
+        write_file_paths(sesh, values, ensemble, date_range, region)
+
+
+def write_file_paths(sesh, variables, ensemble, date_range, area):
+    """Given a variable name write the required file's path to file_paths.txt
+    by querying the CE backend.
+    """
+    logger.info("")
+    logger.info("Translating variables for query")
+    query_args = translate_args(
+        variables["variable"],
+        variables["time_of_year"],
+        variables["temporal"],
+        variables["spatial"],
+        variables["percentile"],
+        area,
+        date_range,
+        ensemble,
+    )
+
+    logger.info("Collecting models")
+    models = get_models(sesh, variables["percentile"], ensemble)
+
+    var_name = "_".join(
+        [
+            variables["variable"],
+            variables["time_of_year"],
+            variables["temporal"],
+            variables["spatial"],
+            variables["percentile"],
+        ]
+    )
+
+    logger.info("Fetching data for {}".format(var_name))
+
+    paths = []
+    [paths.extend(query_backend(sesh, model, query_args)) for model in models]
```

Comment:
It doesn't have to be done in a single line, but we _should_ have the paths be a `set` rather than a `list`. A set will only store unique items. Without that we will have a **massive** number of duplicates.